src/bin/flux-lsp.rs

```diff
@@ -86,7 +86,20 @@ async fn main() {
     let stdout = tokio::io::stdout();
 
     let (service, messages) = LspService::new(|_client| {
-        LspServer::new(!disable_folding, influxdb_url, token, org)
+        let mut server = LspServer::new();
+        if disable_folding {
+            server = server.disable_folding();
+        }
+        if let Some(value) = influxdb_url {
+            server = server.with_influxdb_url(value);
+        }
+        if let Some(value) = token {
+            server = server.with_token(value);
+        }
+        if let Some(value) = org {
+            server = server.with_org(value);
+        }
+        server
```

Comment:
This isn't quite what I think of as the Builder Pattern since I would have expected a separate type called `LspServerBuilder` that has methods like `with_influxdb_url()` etc, plus a method called `build()` which would return a new instance of `LspServer`.

The design you have here sort of implies that you could create an LSP server, use it for a while, call `with_influxdb_url()` to get another server, and then continue to use both servers. I don't think this is the intention?

With a separate `LspServerBuilder` type you might still use a constructor like `LspServer(true, None, ...)` but at least that constructor would only need to be called in a place that is tightly coupled (but separate) from the server itself.

You could also define custom enums to address the issue of boolean arguments that are hard to read.

Reply:
I've never thought the builder pattern explicitly needed a builder, but yeah, this does seem to read a little funny. Ideally, it's something closer to:

```rust
let server = LspServer::new()
                 .with_org("my-org")
                 .with_token("abcdef")
```

I think this makes it read closer to how `tokio`'s server type apis read.